File: vulms_dart/lib/src/models/session.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'session.freezed.dart';

/// Result of a login attempt.
@freezed
abstract class LoginResult with _$LoginResult {
  const factory LoginResult({
    required bool success,
    String? error,
  }) = _LoginResult;
}

/// ASP.NET web form data required for PostBack requests.
@freezed
abstract class AspNetFormData with _$AspNetFormData {
  const factory AspNetFormData({
    required String viewState,
    required String eventValidation,
    String? viewStateGenerator,
    String? previousPage,
  }) = _AspNetFormData;
}

/// Internal session state.
@freezed
abstract class SessionState with _$SessionState {
  const factory SessionState({
    @Default(false) bool isValid,
    String? cookies,
    String? username,
  }) = _SessionState;
}

File: src/types/session.ts
import { z } from 'zod';

export const AspNetWebFormDataSchema = z.object({
  __VIEWSTATE: z.string(),
  __EVENTVALIDATION: z.string(),
  __VIEWSTATEGENERATOR: z.string().optional(),
  __PREVIOUSPAGE: z.string().optional(),
});

export type AspNetWebFormData = z.infer<typeof AspNetWebFormDataSchema>;

export interface SessionState {
  cookies: string;
  isValid: boolean;
  username?: string;
}

export interface LoginResult {
  success: boolean;
  error?: string;
}

export interface ExportedSession {
  username?: string;
  cookies: string;
  createdAt: number;
  exportedAt: number;
  sdkVersion: string;
}

export interface SessionValidationResult {
  valid: boolean;
  reason?: string;
}
